client/src/commands.ts

```diff
@@ -17,16 +17,22 @@ import {
 import * as tasks from "./tasks";
 import type { DenoExtensionContext } from "./types";
 import { WelcomePanel } from "./welcome";
-import { assert } from "./util";
+import { assert, getDenoCommandAndVersion } from "./util";
+import { registryState } from "./lsp_extensions";
+import { createRegistryStateHandler } from "./notification_handlers";
 
+import * as semver from "semver";
 import * as vscode from "vscode";
-import { LanguageClient } from "vscode-languageclient/node";
+import { LanguageClient, ServerOptions } from "vscode-languageclient/node";
 import type {
   DocumentUri,
   Location,
   Position,
 } from "vscode-languageclient/node";
 
+/** The minimum version of Deno that this extension is designed to support. */
+const SERVER_SEMVER = ">=1.11.2";
```

Comment:
I moved a lot of code in here from extension.ts so that it can better handle error paths and do an actual reset when running the `Restart Language Service`.